File: Client/src/modules/models/bullet/ChasingBullet.js
var ChasingBullet = Bullet.extend({
    ctor:function(tower){
        this._super(tower);
    },
    updateBullet: function (){
        var dt = 1/GC.BATTLE.TICK_PER_SECOND;
        //cc.log("bulletLogicPoint: "+this.logicPoint.x + " " +this.logicPoint.y);
        var monsterRadius;
        if (!Helper.checkNull(this.targetMonster) && this.targetMonster.isAlive){
            this.targetLogicPoint = this.targetMonster.getLogicPoint();
            monsterRadius = this.targetMonster.hitRadius;
        }
        else {
            monsterRadius = 0;
        }
        if (Helper.getDistance(this.logicPoint,this.targetLogicPoint) - monsterRadius < this.speed*dt){
            this.logicPoint = this.targetLogicPoint;
            this.bulletUI.toTarget(this.logicPoint);
            if (!Helper.checkNull(this.targetMonster) && this.targetMonster.isAlive){
                this.targetMonster.hitBullet(this.damage);
            }
            this.isAlive = false;
            //TODO something
        }
        else {
            //cc.log("speed + dt: "+this.speed + " "+dt);
            this.logicPoint = Helper.pointMoveToPoint(this.logicPoint,this.targetLogicPoint,this.speed,dt);
            this.bulletUI.updateMapPixel(this.logicPoint);
        }
        /*
        cc.log("ChasingBullet battleObjectId: "+this.battleId);
        cc.log("ChasingBullet curPos: " + this.logicPoint.x + " " + this.logicPoint.y);
        cc.log("ChasingBullet isAlive: "+this.isAlive);

         */

    },

})

File: Server/src/model/Bullet/ChasingBullet.java
package model.Bullet;

import model.Define.ModelDefine;
import model.Tower.AttackTower;

public class ChasingBullet extends Bullet {
    public ChasingBullet(AttackTower tower){
        super(tower, ModelDefine.BULLET_TYPE_CHASING);
    }
    public void update(double dt){
        super.update(dt);
        double monsterRadius;
        if (this.targetMonster != null && this.targetMonster.getIsAlive()){
            this.targetPos = this.targetMonster.getCurPos();
            monsterRadius = this.targetMonster.getHitRadius();
        }
        else {
            monsterRadius = 0;
        }

        if (this.curPos.getDistance(this.targetPos) - monsterRadius < this.speed*dt){
            this.curPos = this.targetPos;
            if (this.targetMonster != null && this.targetMonster.getIsAlive()){
                this.targetMonster.hit(this.damage);
            }
            this.isAlive = false;
        }
        else {
            this.curPos.translate(this.targetPos,this.speed,dt);
        }

//        System.out.println("ChasingBullet battleObjectId: "+this.battleObjectId);
//        System.out.println("ChasingBullet curPos: " + this.curPos.x + " " + this.curPos.y);
//        System.out.println("ChasingBullet isAlive: "+this.isAlive);

    }
}
